include/extractor/guidance/turn_lane_types.hpp

```diff
@@ -0,0 +1,90 @@
+#ifndef OSRM_GUIDANCE_TURN_LANE_TYPES_HPP_
+#define OSRM_GUIDANCE_TURN_LANE_TYPES_HPP_
+
+#include <bitset>
+#include <cstddef>
+#include <cstdint>
+#include <string>
+#include <vector>
+
+#include <boost/assert.hpp>
+#include <boost/functional/hash_fwd.hpp>
+
+#include "util/typedefs.hpp"
+#include "util/simple_logger.hpp"
+
+namespace osrm
+{
+namespace extractor
+{
+namespace guidance
+{
+
+namespace TurnLaneType
+{
+namespace detail
+{
+const constexpr std::size_t num_supported_lane_types = 11;
+} // namespace detail
+
+typedef std::uint16_t Mask;
+const constexpr Mask empty = 0;
+const constexpr Mask none = 1 << 0;
```

Comment:
When using bitshift operators it is good to use the unsigned post-fixes for numbers to make sure we don't accidentally introduce undefined behavior by shifting into the sign bit. Like: `1u << 0u`